middlewares/authenticator.go

```diff
@@ -64,9 +65,20 @@ func NewAuthenticator(authConfig *types.Auth) (*Authenticator, error) {
 	return &authenticator, nil
 }
 
-func parserBasicUsers(users types.Users) (map[string]string, error) {
+func parserBasicUsers(basic *types.Basic) (map[string]string, error) {
+	var userStrs []string
+	if basic.UsersFile != "" {
```

Comment:
If you look at where this function is invoked, `basic` is known not to be `nil` before this function is called.